File: chat/src/main/kotlin/chatox/chat/api/response/ChatRoleResponse.kt
package chatox.chat.api.response

import chatox.chat.model.ChatFeatures
import java.time.ZonedDateTime

data class ChatRoleResponse(
        val id: String,
        val name: String,
        val chatId: String,
        val default: Boolean,
        val templateId: String? = null,
        val features: ChatFeatures,
        val createdAt: ZonedDateTime,
        val createdBy: UserResponse? = null,
        val updatedAt: ZonedDateTime? = null,
        val updatedBy: UserResponse? = null,
        val level: Int
)

File: events/src/chat-roles/types/ChatRoleResponse.ts
import {ChatFeatures} from "./ChatFeatures";
import {User} from "../../common/types";

export interface ChatRoleResponse {
    id: string,
    name: string,
    level: number,
    chatId: string,
    default: boolean,
    features: ChatFeatures,
    createdAt: string,
    createdBy?: User,
    updatedAt?: string,
    updatedBy?: User,
}